### DISCUSSION TITLE
How do I select the first element found by a lens?

### DISCUSSION THREAD

**Discussion by JackWolfard** [yamafaktory/jql#205] [Q&A] [answered]
Hi, given the following JSON and query `"casks".[0]."artifacts"|={"binary"}`

```json
{
  "formulae": "..."
  "casks": [
    {
      "...",
      "artifacts": [
        "...",
        {
          "binary": [
            "/opt/homebrew/Caskroom/miniconda/base/condabin/conda"
          ]
        }
      ]
    }
  ]
}
```

I get the following which is expected

```json
[
  {
    "binary": [
      "/opt/homebrew/Caskroom/miniconda/base/condabin/conda"
    ]
  }
]
```

How do I extract the value `"/opt/homebrew/Caskroom/miniconda/base/condabin/conda"`? When I add `.[0]."binary"`, I get `Root element is not an array`. I'm using the lens feature to filter for all objects which contain the key `"binary"` and then expect to choose the first object if found and extract the path.

**Comment by yamafaktory** [comment, marked as answer]
Hi @JackWolfard ,

What about using simple like `'"casks"[0]"artifacts"[1]"binary"[0]'`?

**Reply by JackWolfard** [reply]
That's a good point. I wanted to avoid hardcoding in a specific ordering of the object incase they weren't stable. That's why I thought lens could be useful since you can filter for objects which have specific keys.

**Comment by yamafaktory** [comment]
@JackWolfard I totally agree with you. I was playing with something similar to your solution and I faced some issues while mixing filters and lenses. This is actually a bug / limitation. I've started to work on a new big refactoring of the project since I think the code has grown in a too complicated way.

**Comment by yamafaktory** [comment]
Incoming changes are on their way :) https://github.com/yamafaktory/jql/pull/206
